tests/beacon/state_machines/test_demo.py

```diff
@@ -1,17 +1,74 @@
+import pytest
+
+
+from eth._utils import bls as bls
 from eth.beacon.db.chain import BeaconChainDB
-from eth.beacon.types.blocks import BaseBeaconBlock
-from eth.beacon.types.states import BeaconState
+from eth.beacon.enums import (
+    SignatureDomain,
+)
+from eth.beacon.helpers import (
+    get_beacon_proposer_index,
+)
+from eth.beacon.state_machines.forks.serenity.blocks import (
+    SerenityBeaconBlock,
+)
+from eth.beacon.types.proposal_signed_data import ProposalSignedData
 
 
-def test_demo(base_db, sample_beacon_block_params, sample_beacon_state_params, fixture_sm_class):
+@pytest.mark.parametrize(
+    (
+        'num_validators,'
+        'epoch_length,'
+        'min_attestation_inclusion_delay,'
+        'target_committee_size,'
+        'shard_count'
+    ),
+    [
+        (10, 2, 1, 2, 2)
+    ]
+)
+def test_demo(base_db,
+              sample_beacon_block_params,
+              genesis_state,
+              fixture_sm_class,
+              config,
+              privkeys,
+              pubkeys):
     chaindb = BeaconChainDB(base_db)
-    state = BeaconState(**sample_beacon_state_params)
-    block = BaseBeaconBlock(**sample_beacon_block_params).copy(
-        slot=state.slot + 2,
+    state = genesis_state
+    block = SerenityBeaconBlock(**sample_beacon_block_params).copy(
+        slot=state.slot + 1,
```

Comment:
I had `+ 2` here to quickly demo that the state transition would process up to the block slot.
I think it is worth leaving that sanity check in here for the time being